Error-Messages.md

```diff
@@ -309,4 +316,226 @@ Integer i1 = null;
+  void requires(){
+
+  }
+}
+```
+
+### for @RequiresNonnull/@EnsuresNonnull annotation, cannot find instance field in class
+This error indicates that the annotation parameter is violating the following syntax rule:
+```
+All parameters given in the annotation must be one of the fields of the class or its super classes.
+```
+Please see the example below where this error will be reported.
+
+```java
+class Super {
+  Object bar;
+}
+
+class C extends Super{
+
+  Object foo;
+  
+  @EnsuresNonnull("foo") // no error: field is present in "C"
+  void ensures(){
+    foo = new Object();
+  }
+
+  @RequiresNonnull("bar") // no error: field is present in super class "Super"
+  void requires(){
+    //not important
+  }
+
+  @RequiresNonnull("tmp") // error: field is not present in this class or any of it's super classes.
+  void requires(){
+    //not important
+  }
+}
+```
+
+### postcondition inheritance is violated, this method must guarantee that all fields written in the @EnsuresNonNull annotation of overridden method are @NonNull at exit point as well. Fields must explicitly appear as parameters at this method @EnsuresNonNull annotation
+
+This error indicates that the annotation violating the following inheritance rule:
+```
+Every method must satisfy all postcondition of its super method.
+```
+
+All overriding methods can only add new field names to the set of fields of its super method
+mentioned in `EnsuresNonNull`.
+
+Please see the example below where this error will be reported.
+```java
+class Super {
+  Object bar;
+  @EnsuresNonnull("bar")
+  void ensures(){
+    foo = new Object();
+  }
+}
+
+class C extends Super{
+
+  Object foo;
+  
+  @EnsuresNonnull("foo") // error: method does not satisfy all its super methods post condition. "bar" must be also mentioned explicitly in the annotation parameter.
+  void ensures(){
+    foo = new Object();
+  }
+
+  // error: method does not satisfy all its super methods post condition. "bar" must be also mentioned explicitly in the annotation parameter.
+  void ensures(){
+    
+  }
+
+  // error: method does satisfy all its super methods post condition, however, "bar" must be also mentioned explicitly in the annotation parameter.
+  void ensures(){
+    bar = new Object();
+  }
+
+  // no-error: method satisfies all its super methods post conditions.
+  @EnsuresNonnull({"foo", "bar"})
+  void ensures(){
+    super.ensures();
+    foo = new Object();
+  }
+}
+```
+
+### precondition inheritance is violated, method in child class cannot have a stricter precondition than its closest overridden method, adding @requiresNonNull for fields makes this method precondition stricter
+This error indicates that the annotation violating the following inheritance rule:
+```
+Every method cannot have a stricter precondition than its super method
+```
+No overriding methods can add new field names to the set of fields of its super method mentioned in `RequiresNonNull` annotation.
```

Comment:
Again, this is missing either an explanation as to _why_ this rule exists, or a link to previous documentation explaining that.